wandb/sdk/wandb_run.py

```diff
@@ -3669,7 +3669,7 @@ def finish(exit_code: Optional[int] = None, quiet: Optional[bool] = None) -> Non
         wandb.run.finish(exit_code=exit_code, quiet=quiet)
 
 
-class InvalidArtifact:
+class UnloggedArtifact(ArtifactInterface):
```

Comment:
what is your sense of moving the artifact logic out of the wandb_run file into a dedicate artifact file?

Reply:
I would prefer to move it elsewhere. Note my other big refactoring PR. Next move is to take the code that's here and push it out, also all the storage handlers.

Reply:
I would prefer to move it elsewhere. Note my other big refactoring PR. Next move is to take the code that's here and push it out, also all the storage handlers.